gamemode/modules/base/sh_util.lua

```diff
@@ -252,24 +252,92 @@ end
 --[[---------------------------------------------------------------------------
 Initialize Physics, throw an error on failure
 ---------------------------------------------------------------------------]]
+local mdlcache = {}
+
 function DarkRP.ValidatedPhysicsInit(ent, solidType)
     solidType = solidType or SOLID_VPHYSICS
 
     if ent:PhysicsInit(solidType) then return true end
 
-    local class, model = "", ""
+    local err, errHints
+    if solidType == SOLID_BSP then
+        err = string.format("%s has no physics and will be motionless", ent)
+        errHints = {
+            "Is this a brush model? SOLID_BSP physics cannot initialize with studio models",
+            "Otherwise, the physics limit was hit"
+        }
+    elseif solidType == SOLID_VPHYSICS then
+        local mdl = string.lower(ent:GetModel() or "")
+
+        if mdl == "" then
+            err = string.format("%s has no model or physics", ent)
+            errHints = {
+                "Did you specify a model for your entity?"
+            }
+        else
+            -- Cache missing model states to prevent repetitive disk accesses
+            -- 0 = no model file
+            -- 1 = no model physics
+            -- 2 = has model physics, hit physics limit
+            local mdlstate = mdlcache[mdl]
+            local checkmdl = mdlstate == nil
+
+            if checkmdl then
+                if file.Exists(mdl, "GAME") then
+                    if util.IsValidProp(mdl) then
+                        mdlexists = 2
+                    else
+                        mdlexists = 1
+                    end
+                else
+                    mdlexists = 0
+                end
+
+                mdlcache[mdl] = mdlexists
+            end
 
-    if IsValid(ent) then
-        class, model = ent:GetClass(), string.format(" (%s)", ent:GetModel() or "Entity has no model at all!")
+            ::ModelError::
+
+            if mdlstate == 0 then
+                err = string.format("%s has missing model \"%s\" and will be invisible and motionless", ent, mdl)
+                errHints = {
+                    "Did you type the model name correctly?",
+                    "Is the model from an addon that is not installed on the server?",
+                    "Is the model from a game that isn't properly mounted? E.g. Counter Strike: Source"
+                }
+            elseif mdlstate == 1 then
+                err = string.format("%s has model \"%s\" with no physics and will be motionless", ent, mdl)
+                errHints = {
+                    "Are you using a viewmodel instead of a worldmodel?",
```

Comment:
This hint is very specific for weapons. It would be confusing if you were to get this hint for e.g. spawned money.

Reply:
I'll just move the hint down to give priority to the more common case (no physics model).